docs/TFM/projectTool/Walkthrough.md

```diff
@@ -1,35 +1,57 @@
+- https://gitforwindows.org/  
   
 ## Set gitbash in the vscode  
-- you may configure the ~/.bashrc & ~/.aliases with your preferences.
+- you may configure the ~/.bashrc & ~/.aliases with your preferences. To accomplish this, you may need to create a .bashrc file. Do that by typing the following commands
+```
+touch ~/.bashrc # creating .bashrc file
+code ~/.bashrc # open file in vs code for editing
+```
+- You can also explore an alternative option that doesn't require editing the ~/.bashrc file. In Visual Studio Code, navigate to "Terminal -> New Terminal," and once the terminal is open, click on the "plus" icon in the terminal window and choose the "Git Bash" option.
   
 ## Download Modelsim  - a system Verilog compiler & simulator (lite free version)  
 - https://www.intel.com/content/www/us/en/software-kit/660907/intel-quartus-prime-lite-edition-design-software-version-20-1-1-for-windows.html  
 Download indevidual files: modelsim, quartus, "Intel MAX 10 Device Support".  
   - Note: after all three programs are downloaded, run the Quartus installation, which will automatically install Modelsim & MAX10.  
 
-## Download the RISCV ToolChain:  
-- https://github.com/xpack-dev-tools/riscv-none-embed-gcc-xpack/releases/ -> File name: "xpack-riscv-none-embed-gcc-10.1.0-1.1-win32-x64.zip")  
+## Download the RISCV ToolChain: 
+In essence, the RISC-V GCC toolchain comprises a suite of software programs designed to transform a C source file into machine code suitable for loading and execution on RISC-V CPUs. This process involves several stages that are not described here. 
+- https://github.com/xpack-dev-tools/riscv-none-embed-gcc-xpack/releases/ -> File name: "xpack-riscv-none-embed-gcc-10.2.0-1.2-win32-x64.zip"). If newer versions are available, you can also download them and then extract into the correct location as shown in the next section.
 - https://xpack.github.io/riscv-none-embed-gcc/install/  -> follow "Manual install" (Only extract in the correct location)  
 
 
 
-## Gitbash shell - compile & link commands:  
+## Gitbash shell - adding to PATH and test installation:  
 Add to the ~/.bashrc the PATH of the gcc compiler
 Example:
 ```
 export PATH=$PATH:/c/Users/abendavid/AppData/Roaming/xPacks/xpack-riscv-none-embed-gcc-10.2.0-1.2/bin
 ```
+To confirm the successful installation, please enter the following command: "riscv-none-embed-gcc --version". If you receive the version information, everything is set up correctly. If not, please revisit the previous steps carefully. 
 
 
-## Test the RISCV toolchain:  
-### 1. Create a simple c program. - "alive.c"  
+## Test the RISCV toolchain:
+
+### 1. Cloning the fpga_mafia environment
+- Create a folder to host your environment and clone fpga_mafia
+```
+cd ~    # you can choose any other reasonable location
+mkdir workspace 
+cd workspace
+git clone https://github.com/amichai-bd/fpga_mafia.git
```

Comment:
My Intention is that the toolchain should be "enabled" without the dependency of the mafia project.
We will refine this later on